Punishment XP and item loss

Vinniezorp opened the pull request:
Death reduces earned XP and all food, drink, medicinal and ammo items by 20%.

Vinniezorp committed:
On player death, Player should lose 20% of their xp.

Vinniezorp committed:
rounded values to prevent overflow when triggering xp loss

Vinniezorp committed:
Death now reduces XP, food, drink, medicine, ammo by 20%

burridgedan commented:
I would suggest maybe adding a check for if the player is a Zombie player as Zombies don't use the inventory so would have nothing to remove anyway. They could still lose EXP though.

`if(player[playerid][iszombie] == 1)`

Secondly, this might be better served having its own function inside of functions.pwn and then calling it in OnPlayerDeath.

Otherwise love the work done - and it resolves #5 :)

Vinniezorp commented:
Thanks for feedback, will correct. Also in retrospect, i'm thinking it would be good to send message to player in the form of "You regain consciousness. You feel weakened and your backpack feels lighter" or something.

burridgedan commented:
Yeah that would work. :)

Vinniezorp committed:
Put code into separate function and call it out in PlayerOnDeath(). Add 5 random lines to indicate item and or xp loss for both zombies and humans.

Vinniezorp commented:
Added the zed check, moved it into its own function and added the player messages as well. Currently 5 rp-lines to be pulled at random for both zombie and human.

burridgedan reviewed (approved):
Love the added flair with the rp lines! :)

I'll merge this into the main repo now, many thanks for taking on #5 :)
